### PR TITLE
Fix maps using incorrect :test function.

### PR DESCRIPTION
**Pull request by bkuehnert**
Now, the underlying lisp hash-table will use 'eq' if the key type is
integral, or 'equal' if the key type is string. This is more efficient
in the integral case, and works properly in the string case (eql doesn't
work for string equality).

Due to some weirdness in the definer macros, this requires the clear-*
function to be defined in the make-map-accessors defun. Admittedly, this
is also due to me doing a poor job when implementing map types, but I
plan to fix this.

### CODE REVIEW

File: define-proto.lisp [hunk 1]
```diff
@@ -593,14 +595,16 @@ Arguments:
...
-        (declaim (inline ,clear-function-name))
-        (defun ,clear-function-name (,obj)
-          ,(if bool-index
-               `(setf (bit (,bit-field-name ,obj) ,bool-index)
-                      ,(if default-form 1 0))
-               `(setf (,hidden-accessor-name ,obj) ,default-form))
-          ,(when index
-            `(setf (bit (,is-set-accessor ,obj) ,index) 0)))
+        ;; Map type clear functions are created in make-map-accessor-forms.
+        ,(unless (eq (proto-set-type field) :map)
```
Comment (Slids) [inline define-proto.lisp:599]: add a comment above 
;; todo(bkuehnert): Fix map type so we can craete the clear functions here.

File: tests/map-test.lisp [hunk 2]
```diff
@@ -62,6 +62,15 @@
     (assert-equal (map-all.msgmap-gethash 0 test) nil)
     (assert-equal (map-all.enummap-gethash 0 test) :one)))
 
+;; Verify that the lisp hash-table properly handles string keys
+(deftest string-key-check (map-suite)
```
Comment (Slids) [inline tests/map-test.lisp:66]: Add a mixed case test, like
"Foo"
and "foo"

File: define-proto.lisp (hunk 1 above)
Comment (bkuehnert) [reply define-proto.lisp:599]: Fixing this also requires removing the field-data struct, so it's not *all* my fault. Adding a comment anyway.

File: define-proto.lisp [hunk 3]
```diff
@@ -593,14 +595,17 @@ Arguments:
...
-          ,(if bool-index
-               `(setf (bit (,bit-field-name ,obj) ,bool-index)
-                      ,(if default-form 1 0))
-               `(setf (,hidden-accessor-name ,obj) ,default-form))
-          ,(when index
-            `(setf (bit (,is-set-accessor ,obj) ,index) 0)))
+        ;; Map type clear functions are created in make-map-accessor-forms.
+        ;; todo(benkuehnert): rewrite map types/definers so that this isn't necessary
+        ,(unless (eq (proto-set-type field) :map)
+           `(declaim (inline ,clear-function-name))
```
Comment (cgay) [inline define-proto.lisp:601]: This declaim form is dropped on the floor. Only the value of the last form in the unless becomes part of the returned list.

**Pull request closed without merging**